docs/source/tutorials/arch_modeling/Spypads_tutorial.rst

```diff
@@ -0,0 +1,334 @@
+Creating Spypads Using XML Syntax
+=================================
+
+Introduction
+~~~~~~~~~~~~
+
+**In this tutorial, we will**
+   - Show the XML Syntax for Global Outputs
+   - Showcase an example with Spypads
+   - Modify an existing architecture to incorporate Spypads
+   - Verify correctness through GTKWave
+
+Through this tutorial, we will show how to create Spypads in OpenFPGA.
+
+Spypads are physical output pins on a FPGA chip through which you can read out internal signals when doing silicon-level debugging. The XML syntax for spypads and other 
+global signals can be found on our :ref:`circuit_library` documentation page.
+
+To create a spypad, the ``port type`` needs to be set to **output** and ``is_global`` and ``is_io`` need to be set to **true**:
+
+.. code-block:: xml
+
+   <port type="output" is_global="true" is_io="true"/>
+
+When the port is syntactically correct, the outputs are independently wired from different instances to separated FPGA outputs and would physically look like :numref:`fig_gpout`:
+
+.. _fig_gpout:
+
+.. figure:: ./figures/gpout_ports.png

```

Comment:
You do not need to duplicate the figure. You can simply refer to an existing figure using :ref:``